test/Reader/Http/Psr7ResponseDecoratorTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace ZendTest\Feed\Reader\Http;
+
+use PHPUnit_Framework_TestCase as TestCase;
+use Psr\Http\Message\ResponseInterface as Psr7ResponseInterface;
+use Zend\Feed\Reader\Http\Psr7ResponseDecorator;
+use Zend\Feed\Reader\Http\HeaderAwareResponseInterface;
+use Zend\Feed\Reader\Http\ResponseInterface;
+use ZendTest\Feed\Reader\TestAsset\Psr7Stream;
+
+class Psr7ResponseDecoratorTest extends TestCase
+{
+    public function testDecoratorIsAFeedResponse()
+    {
+        $originalResponse = $this->prophesize(Psr7ResponseInterface::class);
+        $decorator = new Psr7ResponseDecorator($originalResponse->reveal());
+        $this->assertInstanceOf(ResponseInterface::class, $decorator);
+    }
+
+    public function testDecoratorIsAHeaderAwareResponse()
+    {
+        $originalResponse = $this->prophesize(Psr7ResponseInterface::class);
+        $decorator = new Psr7ResponseDecorator($originalResponse->reveal());
+        $this->assertInstanceOf(HeaderAwareResponseInterface::class, $decorator);
+    }
+
+    public function testDecoratorIsNotAPsr7Response()
+    {
+        $originalResponse = $this->prophesize(Psr7ResponseInterface::class);
+        $decorator = new Psr7ResponseDecorator($originalResponse->reveal());
+        $this->assertNotInstanceOf(Psr7ResponseInterface::class, $decorator);
+    }
+
+    public function testCanRetrieveDecoratedResponse()
+    {
+        $originalResponse = $this->prophesize(Psr7ResponseInterface::class);
+        $decorator = new Psr7ResponseDecorator($originalResponse->reveal());
+        $this->assertSame($originalResponse->reveal(), $decorator->getDecoratedResponse());
+    }
+
+    public function testProxiesToDecoratedResponseToRetrieveStatusCode()
+    {
+        $originalResponse = $this->prophesize(Psr7ResponseInterface::class);
+        $originalResponse->getStatusCode()->willReturn(301);
+        $decorator = new Psr7ResponseDecorator($originalResponse->reveal());
+        $this->assertSame(301, $decorator->getStatusCode());
+    }
+
+    public function testProxiesToDecoratedResponseToRetrieveBody()
+    {
+        $originalResponse = $this->prophesize(Psr7ResponseInterface::class);
+        $originalResponse->getBody()->willReturn('BODY');
+        $decorator = new Psr7ResponseDecorator($originalResponse->reveal());
+        $this->assertSame('BODY', $decorator->getBody());
+    }
+
+    public function testCastsStreamToStringWhenReturningPsr7Body()
+    {
+        $stream = new Psr7Stream('BODY');
```

Comment:
> I suggest change Psr7Stream as any standard class with the __toString method.

I was going to raise an objection that this won't work, as it breaks the contract, but internally, we're only returning the results of calling `getBody()`, and since PHP versions less than 7 don't support return type hinting, we can return anything here and not make the code break. Updating now.